web/src/main/java/com/google/accompanist/web/WebView.kt

```diff
@@ -0,0 +1,164 @@
+ * the WebView back.
+ * @param onCreated Called when the WebView is first created, this can be used to set additional
+ * settings on the WebView.
+ *
+ * @sample com.google.accompanist.sample.webview.BasicWebViewSample
+ */
+@Composable
+fun WebView(
+    state: WebViewState,
+    onContentChanged: (WebContent) -> Unit,
+    modifier: Modifier = Modifier,
+    captureBackPresses: Boolean = true,
+    onCreated: (WebView) -> Unit = {}
+) {
+    var view by remember { mutableStateOf<WebView?>(null) }
+    var webViewLoaded by remember(view) { mutableStateOf(false) }
+
+    BackHandler(captureBackPresses && state.canGoBack) {
+        view?.goBack()
+    }
+
+    AndroidView(
+        factory = {
+            WebView(it).apply {
+                onCreated(this)
+
+                webViewClient = object : WebViewClient() {
+                    override fun onPageStarted(view: WebView?, url: String?, favicon: Bitmap?) {
+                        super.onPageStarted(view, url, favicon)
+                        state.isLoading = true
+
+                        // When a WebView is first created it sends a page started event
+                        // with about:blank. We don't want to forward this on.
+                        // It's important we still support a link to about:blank though.
+                        if (webViewLoaded || url != "about:blank") {
+                            onContentChanged(WebContent.Url(Uri.parse(url)))
+                            webViewLoaded = true
+                        }
+                    }
+
+                    override fun onPageFinished(view: WebView?, url: String?) {
+                        super.onPageFinished(view, url)
+                        state.isLoading = false
+                    }
+
+                    override fun shouldOverrideUrlLoading(
+                        view: WebView?,
+                        request: WebResourceRequest?
+                    ): Boolean {
+                        // Override all url loads to make the single source of truth
+                        // of the URL the state holder Url
+                        request?.let {
+                            val content = WebContent.Url(it.url)
+                            onContentChanged(content)
+                        }
+                        return true
+                    }
+                }
+            }.also { view = it }
+        },
+        modifier = modifier
+    ) { webView ->
+        when (val l = state.content) {
+            is WebContent.Url -> {
+                val url = l.uri.toString()
+
+                if (url.isNotEmpty() && url != webView.url) {
+                    webView.loadUrl(url)
+                }
+            }
+            is WebContent.Data -> {
+                webView.loadDataWithBaseURL(l.baseUrl, l.data, null, null, null)
+            }
+        }
+
+        state.canGoBack = webView.canGoBack()
+    }
+}
+
+sealed class WebContent {
```

Comment:
Can this be expanded in the future? If so, changes would provoke breaking changes

Reply:
I can't think of any other cases